doc/requirements/Requirements.md

```diff
@@ -0,0 +1,704 @@
+<why>To catch user out of valid range errors when changing the parameter
+value.</why>
+
+##### Bit field following the C ABI
+PF **SHOULD** support 1 or more bit sized integers.
+<why>The reference implementation supports it.</why>
+
+Such parameters called are regrouped in a so called bit parameter block.
+The API to access a bit parameter block **SHOULD** give access to a packed bit
+field.
+<why>The reference implementation supports it.</why>
+
+### Parameter adaptation
+
+#### Philosophy
+Parameters exposed by hardware sometimes need to transform a parameter value.
+For example an hardware integer parameter could have a range 64-128 but it might
+be necessary for upper layer to access in a range 0-100.
+
+This transformation (called parameter adaptation in the rest of the document)
+could be done by the syncer. Nevertheless syncers are supposed to contain only
+business logic and should not be impacted by upper layer needs.
+
+#### Definition
+<dl>
+<dt>Parameter adaptation<dt>
+<dd>
+A bijective pure function converting a parameter value between the syncer
+and other parameter reader/writer (including the inference engine).
+
+<why>It must be bijective in order to a) scale the user value to the hardware
+value and b) convert the hardware value to the user's value space.)</why>
+</dd>
+</dl>
+
+#### Requirements
+The following parameter adaptation **SHOULD** be supported
+
+ - Affine adaptation: `affAd(value) = slope * value + offset` where slope and
+   offset and user-defined constants
+   <why>The reference implementation supports it.</why>
+
+ - Logarithm adaptation: `logAd(base, value) = ln(value) / ln(base)` where
+   `ln` is the natural logarithm and base is a user-defined constant.
+   <why>The reference application supports it.</why>
+   <nb>The reference implementation also supports passing a floor value to be
+   applied after conversion.</nb>
+
+A PF **MAY** offer Parameter adaptation composition. I.e. combine multiple parameter
+adaptation
+<nb>E.g.: composing the affine and logarithm adaptation to
+`compAd(value) = slope * logAd(base, value) + offset`.</nb>
+<why>To avoid combination explosion of parameter adaptations. The idea is to
+builtin basic function and let the user compose them to meet its need.</why>
+<ko>The reference application supports in a tricky way: the logarithm
+adaptation is always combined with the affine adaptation</ko>
+
+## Identifiers
+Every parameter **MUST** have an identifier that uniquely identifies it.
+<why>to identify a parameter outside the framework</why>
+
+This identifier **SHOULD** be a string.
+<why>so that the (human</why> user can identify a parameter with ease.)
+
+Two PF instances with the same parameters **MUST** have the same identifier for
+those parameters.
+I.e. this identifier should be the same across all instances with the same
+configuration.
+<why>Persistence of parameter identifier across PF instances with the same
```

Comment:
as far as I understand, this requirement is more "all instance should share the same format of persistance", am I wrong?